src/plugins/ytdlp.desktop/native.ts

```diff
@@ -0,0 +1,300 @@
+    if (!fs.existsSync(_workdir)) fs.mkdirSync(_workdir, { recursive: true });
+    workdir = _workdir;
+    log("Using workdir: ", workdir);
+    return workdir;
+}
+export async function stop(_: IpcMainInvokeEvent) {
+    if (workdir) {
+        log("Cleaning up workdir");
+        fs.rmSync(workdir, { recursive: true });
+        workdir = null;
+    }
+}
+
+async function metadata(options: DownloadOptions) {
+    stdout_global = "";
+    const metadata = JSON.parse(await ytdlp(["-J", options.url, "--no-warnings"]));
+    if (metadata.is_live) throw "Live streams are not supported.";
+    stdout_global = "";
+    return { videoTitle: `${metadata.title || "video"} (${metadata.id})` };
+}
+function genFormat({ videoTitle }: { videoTitle: string; }, { maxFileSize, format }: DownloadOptions) {
+    const HAS_LIMIT = !!maxFileSize;
+    const MAX_VIDEO_SIZE = HAS_LIMIT ? maxFileSize * 0.8 : 0;
+    const MAX_AUDIO_SIZE = HAS_LIMIT ? maxFileSize * 0.2 : 0;
+
+    const audio = {
+        noFfmpeg: "ba[ext=mp3]{TOT_SIZE}/wa[ext=mp3]{TOT_SIZE}",
+        ffmpeg: "ba*{TOT_SIZE}/ba{TOT_SIZE}/wa*{TOT_SIZE}/ba*"
+    };
+    const video = {
+        noFfmpeg: "b{TOT_SIZE}{HEIGHT}[ext=webm]/b{TOT_SIZE}{HEIGHT}[ext=mp4]/w{HEIGHT}{TOT_SIZE}",
+        ffmpeg: "b*{VID_SIZE}{HEIGHT}+ba{AUD_SIZE}/b{TOT_SIZE}{HEIGHT}/b*{HEIGHT}+ba",
+    };
+    const gif = {
+        ffmpeg: "bv{TOT_SIZE}/wv{TOT_SIZE}"
+    };
+
+    let format_group: { noFfmpeg?: string; ffmpeg: string; };
+    switch (format) {
+        case "audio":
+            format_group = audio;
+            break;
+        case "gif":
+            format_group = gif;
+            break;
+        case "video":
+        default:
+            format_group = video;
+            break;
+    }
+
+    const format_string = (ffmpegAvailable ? format_group.ffmpeg : format_group.noFfmpeg)
+        ?.replaceAll("{TOT_SIZE}", HAS_LIMIT ? `[filesize<${maxFileSize}]` : "")
+        .replaceAll("{VID_SIZE}", HAS_LIMIT ? `[filesize<${MAX_VIDEO_SIZE}]` : "")
+        .replaceAll("{AUD_SIZE}", HAS_LIMIT ? `[filesize<${MAX_AUDIO_SIZE}]` : "")
+        .replaceAll("{HEIGHT}", "[height<=1080]");
+    if (!format_string) throw "Gif format is only supported with ffmpeg.";
+    log("Video formated calculated as ", format_string);
+    log(`Based on: format=${format}, maxFileSize=${maxFileSize}, ffmpegAvailable=${ffmpegAvailable}`);
+    return { format: format_string, videoTitle };
+}
+async function download({ format, videoTitle }: { format: string; videoTitle: string; }, { ytdlpArgs, url, format: usrFormat }: DownloadOptions) {
+    cleanVideoFiles();
+    const baseArgs = ["-f", format, "-o", "download.%(ext)s", "--force-overwrites", "-I", "1"];
+    const remuxArgs = ffmpegAvailable
+        ? usrFormat === "video"
+            ? ["--remux-video", "webm>webm/mp4"]
+            : usrFormat === "audio"
+                ? ["--remux-audio", "mp3"] // @ TODO THIS DOESNT EXIST
```

Comment:
```suggestion
                ? ["--extract-audio", "--audio-format", "mp3"]
```
you can replace it with this and it'll work, however it downloads the video, extracts the audio and then deletes the video

Reply:
Oh crap, good catch. Either it was copilot or my brain fried, because `--remux-audio` is not a real flag lol.

Reply:
> Oh crap, good catch. Either it was copilot or my brain fried, because `--remux-audio` is not a real flag lol.

it was mentioned as a suggestion in a github issue a while ago, perhaps thats why copilot suggested it

Reply:
67bf826